### PR TITLE
Jakarta Batch 2.1 Plan Review

### PR DESCRIPTION
**Pull request by scottkurz**
Signed-off-by: Scott Kurz <skurz@us.ibm.com>

## Plan Review
When creating a specification project plan review, create PRs with the content defined as follows.

Include the following in the PR:
- [X] A directory in the form wombat/x.y where x.y is the release major.minor version.
- [X] An index page wombat/x.y/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md)
- [X] Title should be on the form "Jakarta Wombat X.Y (under development)"
- [X] Includes a plan for the release. Usually, a couple of statements is enough. Otherwise, a link to a document describing the release.
- [X] Includes a detailed description of any backward incompatibility (Mark with **N/A** and check if none)
- [X] Declares optional features if there are any (Mark with **N/A** and check if none)
- [X] Includes minimum Java SE version
- [X] Describe any updates or required changes including splitting out the TCK (Mark with **N/A** and check if not planned)
- [X] Link to updated release record
  https://projects.eclipse.org/projects/ee4j.batch/releases/2.1.0

### CODE REVIEW

**Comment by scottkurz** [comment]
@starksm64 @kwsutter  - I updated the PR checklist and added a section on CDI integration (is this optional or not? - something I've struggled to get my head around).   

I realize the CDI discussion might need to be resolved for going to ballot.   I'll try one of the MLs unless I hear from you first some other forum would be better.   Thanks.

**Comment by starksm64** [comment]
Not that I can see what in this PR might cause it, but the site preview link given above shows a broken landing page. Following the links down it is missing the proper layout at every level, all the way down to the 2.1 batch page. Looking at the page in dev mode shows the styles.css and solstice.js are missing from the resulting build:

[Error] Failed to load resource: the server responded with a status of 404 () (styles.css, line 0)
[Error] Failed to load resource: the server responded with a status of 404 () (solstice.js, line 0)

That must be a build problem on the EF infrastructure side.

In terms of the CDI question, we don't want to have the notion of optional features in specs any longer. So, either the spec defines how to integrate with CDI if batch is run in an environment where CDI is present, or it says nothing.

File: batch/2.1/_index.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+* Any updates required to meet the Java version requirements of Jakarta EE 10
+* Any requirements identified by other specifications or the Jakarta EE 10 platform projects during the release cycle
+
+## Optional Features: 
+
+The integration with CDI deserves some discussion.
+
+The high-level intent is to say that when CDI is present, as within the Jakarta EE Platform, then the integration of Jakarta Batch + CDI will be well-defined and so required for an implementation to be certified as Jakarta Batch-compliant.   However, when CDI is not present, an implementation should be able to certify full Jakarta Batch compliance.
+
+TBD how to phrase this practically, so that it can be implemented by the TCK, for example.   E.g. should this be an optional subset of tests within this TCK or should the relevant tests no-op or behave differently if/when CDI is not detected, (e.g. because `CDI.current()` returns null).
```
Comment (starksm64) [inline batch/2.1/_index.md:40]: We want to move away from optional behavior, so it should be phrased as a requirement for when running batch in CDI aware environments and the TCK can validate that using something like the CDI.current() check you mention.

Comment (scottkurz) [reply batch/2.1/_index.md]: I like that approach better.   Made the change.

**Comment by scottkurz** [comment]
Please consider this on hold for another week.

I'm going to rework the plan review wording around CDI, since we're moving now towards treating CDI integration as a Required function in 2.1.

**Review by starksm64 — APPROVED** [review]

**Comment by starksm64** [comment]
I'll start the ballot today

**Comment by scottkurz** [comment]
@starksm64 sorry, I didn't see you had started the ballot.  (I did leave a note saying I was making another change, but should've put the PR in draft).  I just force pushed over the earlier changes.

Perhaps I got the change in early enough to not affect things?

**Review by starksm64 — APPROVED** [review]

**Comment by starksm64** [comment]
The vote concluded successfully and passed on June 11:

|                       |  Yes    | No  | Abstain  |
|-----------------------|---------|-----|----------|
|Fujitsu                |   +1      |     |          |
|IBM                    |     +1    |     |          |
|Oracle                 |    +1     |     |          |
|Payara                 |    +1     |     |          |
|Red Hat                |   +1      |     |          |
|Tomitribe              |    +1     |     |          |
|EE4J PMC               |    +1     |     |          |
|Participant Members    |   +1      |     |          |
|Committer Members      |     +1    |     |          |
|Enterprise Members |+1|     |          |
| Total|+10|

**Pull request merged**